app/forms/user_form.rb

```diff
@@ -0,0 +1,26 @@
+class UserForm
+
+  include ActiveModel::Model
+
+  attr_accessor :first_name, :last_name, :password, :role, :email
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :password, presence: true, length: { minimum: 8 }
+  validates :role, presence: true
+  validates :email, presence: true, length: { minimum: 8 }
+
+  def initialize(params= {})
+    @user = User.new
+    super(params)
+  end
+
+  def submit
+    return false if invalid?
+    @user.first_name = @first_name,
+    @user.last_name = @last_name,
+    @user.email = @email,
+    @user.role = @role,
+    @user.password = @password,
+    @user.save
+  end
+end
```

Comment:
Your implementation of Form Objects is correct. However, from the perspective of "Clean Code Architecture" we mix different layers here. So, in some project we tend to separate validations and DB updates in different classes.